modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/api/ItSqlApiBaseTest.java

```diff
@@ -315,28 +319,50 @@ public void checkExplicitTxRollback() {
     }
 
     /** Check correctness of rw and ro transactions for table scan. */
-    @Test
-    public void checkMixedTransactionsForTable() {
+    @CartesianTest(name = "roTx={0} commit={1} explicit={2}")
+    public void checkMixedTransactionsForTable(
+            @Values(booleans = {true, false}) boolean roTx,
+            @Values(booleans = {true, false}) boolean commit,
+            @Values(booleans = {true, false}) boolean explicit
+    ) {
         sql("CREATE TABLE TEST(ID INT PRIMARY KEY, VAL0 INT)");
 
         Matcher<String> planMatcher = containsTableScan("PUBLIC", "TEST");
 
-        checkMixedTransactions(planMatcher);
+        checkMixedTransactions(planMatcher, roTx, commit, explicit);
     }
 
 
     /** Check correctness of rw and ro transactions for index scan. */
-    @Test
-    public void checkMixedTransactionsForIndex() throws Exception {
+    @CartesianTest(name = "roTx={0} commit={1} explicit={2}")
+    public void checkMixedTransactionsForIndex(
+            @Values(booleans = {true, false}) boolean roTx,
+            @Values(booleans = {true, false}) boolean commit,
+            @Values(booleans = {true, false}) boolean explicit
+    ) throws InterruptedException {
         sql("CREATE TABLE TEST(ID INT PRIMARY KEY, VAL0 INT)");
         sql("CREATE INDEX TEST_IDX ON TEST(VAL0)");
 
+        // Wait for the index to be created.
+        assertTrue(waitForCondition(
+                () -> CLUSTER.runningNodes().allMatch(node -> {
```

Comment:
Why you should waiting appear index for all nodes, as I understand it should be enough wait just publish state for one node.

Reply:
Can you explain why?

Reply:
publication in the catalog should occur simultaneously for all nodes

Reply:
I don't think that's true but that's not relevant anymore